File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/chat/domain/ChatMessage.java
package io.twogether.nbe_5_7_2_02team.chat.domain;

import io.twogether.nbe_5_7_2_02team.global.common.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ChatMessage extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "chatroom_id")
    private ChatRoom chatRoom;

    @ManyToOne
    @JoinColumn(name = "chatroom_member_id")
    private ChatMember chatMember;

    @Column(name = "content")
    private String content;

    @Builder
    public ChatMessage(ChatRoom chatRoom, ChatMember chatMember, String content) {
        this.chatRoom = chatRoom;
        this.chatMember = chatMember;
        this.content = content;
    }

    public Long getId() {
        return id;
    }

    public ChatRoom getChatRoom() {
        return chatRoom;
    }

    public ChatMember getChatMember() {
        return chatMember;
    }

    public String getContent() {
        return content;
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/chat/domain/ChatMessage.kt
package io.twogether.nbe_5_7_2_02team.chat.domain

import io.twogether.nbe_5_7_2_02team.global.common.BaseEntity
import jakarta.persistence.*

@Entity
data class ChatMessage(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @field:JoinColumn(name = "chatroom_id")
    @field:ManyToOne
    val chatRoom: ChatRoom,
    @field:JoinColumn(name = "chatroom_member_id")
    @field:ManyToOne
    val chatMember: ChatMember,
    @field:Column(name = "content")
    val content: String,
) : BaseEntity()
